activerecord/test/cases/associations/has_one_through_disable_joins_associations_test.rb

```diff
@@ -0,0 +1,416 @@
+    assert new_member.current_membership
+    assert_equal new_club, new_member.club_without_joins
+    assert_predicate new_club, :new_record?
+    assert_predicate new_member.current_membership, :new_record?
+    assert new_member.save
+    assert_predicate new_club, :persisted?
+    assert_predicate new_member.current_membership, :persisted?
+  end
+
+  def test_association_build_constructor_builds_through_record
+    new_member = Member.create(name: "Chris")
+    new_club = new_member.build_club_without_joins
+    assert new_member.current_membership
+    assert_equal new_club, new_member.club_without_joins
+    assert_predicate new_club, :new_record?
+    assert_predicate new_member.current_membership, :new_record?
+    assert new_member.save
+    assert_predicate new_club, :persisted?
+    assert_predicate new_member.current_membership, :persisted?
+  end
+
+  def test_creating_association_builds_through_record_for_new
+    new_member = Member.new(name: "Jane")
+    new_member.club_without_joins = clubs(:moustache_club)
+    assert new_member.current_membership
+    assert_equal clubs(:moustache_club), new_member.current_membership.club
+    assert_equal clubs(:moustache_club), new_member.club_without_joins
+    assert new_member.save
+    assert_equal clubs(:moustache_club), new_member.club_without_joins
+  end
+
+  def test_creating_association_sets_both_parent_ids_for_new
+    member = Member.new(name: "Sean Griffin")
+    club = Club.new(name: "Da Club")
+
+    member.club_without_joins = club
+
+    member.save!
+
+    assert member.id
+    assert club.id
+    assert_equal member.id, member.current_membership.member_id
+    assert_equal club.id, member.current_membership.club_id
+  end
+
+  def test_replace_target_record
+    new_club = Club.create(name: "Marx Bros")
+    @member.club_without_joins = new_club
+    @member.reload
+    assert_equal new_club, @member.club_without_joins
+  end
+
+  def test_has_one_through_polymorphic
+    assert_equal clubs(:moustache_club), @member.sponsor_club_without_joins
+  end
+
+  def test_has_one_through_eager_loading
+    members = assert_queries(3) do # base table, through table, clubs table
+      Member.all.merge!(includes: :club_without_joins, where: ["name = ?", "Groucho Marx"]).to_a
+    end
+    assert_equal 1, members.size
+    assert_not_nil assert_no_queries { members[0].club_without_joins }
+  end
+
+  def test_has_one_through_eager_loading_through_polymorphic
+    members = assert_queries(3) do # base table, through table, clubs table
+      Member.all.merge!(includes: :sponsor_club_without_joins, where: ["name = ?", "Groucho Marx"]).to_a
+    end
+    assert_equal 1, members.size
+    assert_not_nil assert_no_queries { members[0].sponsor_club_without_joins }
+  end
+
+  def test_has_one_through_with_conditions_eager_loading
+    # conditions on the through table
+    member = Member.all.merge!(includes: :favorite_club_without_joins).find(@member.id)
```

Comment:
This fails locally and I wonder if it's not a hint at some part of the implementation I need to change. It fails with:

```
ActiveRecord::StatementInvalid: SQLite3::SQLException: no such column: memberships.favorite
```

However, when I look at activerecord/test/schema/schema.rb, I see the `memberships` table definitely has a `favorite` field.

Reply:
I wonder if this isn't expected because of the queries being separate, actually. Maybe I should try using `preload` instead.